acceptance/tests/resource/service/windows.rb

```diff
@@ -0,0 +1,122 @@
+test_name "Windows Service Provider" do
```

Comment:
It might be useful to add an acceptance test that ensures Puppet fails if the service's pending transition hasn't made any progress since the last wait hint. The C# class can be modified to change the `Sleep` method to something like `StubPendingTransition(waitHint, duration)` and the fixture can take additional parameters like `onStart_waitHint`, `onStart_duration` for each of the `on` methods to get this to work.